File: prop-guice/src/main/java/io/pleo/prop/guice/internal/PropResult.java
package io.pleo.prop.guice.internal;

import io.pleo.prop.core.Prop;

public class PropResult {
  private Prop<?> prop;
  private Throwable error;

  public PropResult(Prop<?> prop) {
    this.prop = prop;
  }

  public PropResult(Throwable error) {
    this.error = error;
  }

  public Prop<?> getProp() {
    return prop;
  }

  public Throwable getError() {
    return error;
  }

  public boolean isError() {
    return error != null;
  }
}

File: prop-guice/src/main/kotlin/io/pleo/prop/guice/internal/PropResult.kt
package io.pleo.prop.guice.internal

import io.pleo.prop.core.Prop

class PropResult {
    var prop: Prop<*>? = null
        private set

    var error: Throwable? = null
        private set

    constructor(prop: Prop<*>?) {
        this.prop = prop
    }

    constructor(error: Throwable?) {
        this.error = error
    }

    fun isError() = error != null
}
